homeassistant/components/flunearyou/diagnostics.py

```diff
@@ -0,0 +1,36 @@
+"""Diagnostics support for Flu Near You."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CATEGORY_CDC_REPORT, CATEGORY_USER_REPORT, DOMAIN
+
+TO_REDACT = {
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinators: dict[str, DataUpdateCoordinator] = hass.data[DOMAIN][entry.entry_id]
+
+    return {
+        "data": {
+            CATEGORY_CDC_REPORT: async_redact_data(
```

Comment:
`async_redact_data` is now called 2 times, that function is recursive so you can use it on the entire dictionary at once